### ISSUE TITLE
Update ESS-DIVE suite

### ISSUE THREAD

**Issue by gothub** [NCEAS/metadig-checks#423]
## Description

Update the ESS-DIVE assessment suite based on the checks described [here](https://docs.google.com/spreadsheets/d/14v3hjPL9jDSgfSF6RCyDgwEKZIK4xwAzJRCb0zjTsfg/edit#gid=1908621675).

The new version number will be 1.1.0
Note that some of these checks are new and some are currently part of the FAIR suite.

List of checks for version 1.1.0:

| Description | Check Id / Issue| Category | Level | Implemented? | Comments |
|---|---|---|---|---|---|
| Project name (title) controlled|[ressource.projectTitle.controlled](https://github.com/NCEAS/metadig-checks/issues/420)|Findable|Optional|Y|new for suite v1.1.0
|Any links included in metadata resolve |[resource.URLs.resolvable](https://github.com/NCEAS/metadig-checks/issues/422) |Accessible|Required|Y|new for suite v1.1.0|
|Dataset Landing Page Resolves|[resource.landingPage.present](https://github.com/NCEAS/metadig-checks/issues/424)| Accessible |Required|Not for ESS-DIVE Suite 1.1.0 suite|same as https://github.com/NCEAS/metadig-checks/issues/188?|
|Existing DOI and Alternate Identifier |[Existing DOI and Alternate Identifier](https://github.com/NCEAS/metadig-checks/issues/427), [metadata.identifier.resolvable](https://github.com/NCEAS/metadig-checks/issues/60) | Accessible |Required| Not for ESS-DIVE Suite 1.1.0 |The DOI portion of this check is already implemented as https://github.com/NCEAS/metadig-checks/issues/60|
|Project funder is controlled|[resource.awardFunderName.controlled](https://github.com/NCEAS/metadig-checks/issues/421) | Findable |Optional| Y |new for suite v1.1.0|
|File formats are not proprietary |[entity.type.nonproprietary](https://github.com/NCEAS/metadig-checks/issues/436) | Interoperable |Optional| Y |new for suite v1.1.0|
|File present |[File present](https://github.com/NCEAS/metadig-checks/issues/429), [check.entity.present](https://github.com/NCEAS/metadig-checks/issues/126)|Interoperable|INFO|Y|new for suite v1.1.0|
|DOI and alternate identifier |[metadata.alternateIdentifier.resolvable](https://github.com/NCEAS/metadig-checks/issues/427)|Findable|Optional|Not for ESS-DIVE Suite 1.1.0||
|Metadata identifier is resolvable|metadata.identifier.resolvable|Accessible|Optional|Y|new for suite v1.1.0|
||check.creator.present|Findable|REQUIRED|Y||
||check.creator.info|Findable|REQUIRED|Y||
||contact.has.ORCID|Findable|REQUIRED|Y||
||check.contact.info|Findable|REQUIRED|Y||
||check.abstract.100.words|Findable|REQUIRED|Y||
||dataset.keywords.minimum|Findable|REQUIRED|Y||
||dataset.keywords.overlap|Findable|OPTIONAL|Y||
||dataset.title.length2|Findable|REQUIRED|Y||
||check.identifier.is.present|Findable|REQUIRED|Y||
||check.usage.is.cc|Reusable|OPTIONAL|Y||
||check.temporal.coverage|Findable|OPTIONAL|Y||
||check.geographic.description|Findable|REQUIRED|Y||
||check.bounding.coordinates|Findable|REQUIRED|Y||
||check.pub.date|Findable|REQUIRED|Y||
||check.methods.present|Interoperable|REQUIRED|Y||

**Comment by gothub** [comment]
@vchendrix has requested that the new suite be version 1.1.0

**Comment by gothub** [comment]
Discussion topics for the ESS-DIVE Suite update

ESS-DIVE Suite
- Can any of these checks that are denoted as new, ESS-DIVE specific checks
  use a modified version of an existing check?

- The ESS-DIVE suite has the check categories ('types') of
  - Identification
  - Discovery
  - Interpretation

- Do you want to start using the FAIR categories? They are:
  - Findable
  - Accessible
  - Interoperable
  - Reusable
- Potential categories for the  checks are listed on the spreadsheet at https://docs.google.com/spreadsheets/d/14v3hjPL9jDSgfSF6RCyDgwEKZIK4xwAzJRCb0zjTsfg/edit#gid=1908621675
- if using FAIR categories, then all other ESS-DIVE checks need to be re-categorized

Checks

- Project name accuracy
  - how often does the project list need to be updated?
  - can this be included with the metadig-checks distribution or does it have to
    be dynamically downloaded from the website (https://data.ess-dive.lbl.gov/js/themes/ess-dive/data/projects.json)?
- Any links included in metadata resolve
  - links in abstract, location description, methods, and related references resolve
    - this check may take awhile to resolve, is that OK?
- Dataset Landing Page Resolves
  - resource.landingPage.present
    - checks /eml/dataset/distribution/online/url[@function="information"]
  - can ESS-DIVE use this check?
- Existing DOI and Alternate Identifier#427
  - does the check verify that the metadata id is a DOI?
  - are 2 checks needed?
  - metadata.identifier.resolvable - existing
  - alternate.identifier.resolvable - new
  - both checks verify that identifier is a valid DOI
- Funding Organizations
  - ESS-DIVE not using EML 2.2 awarad fields
  - /eml/dataset/project/funding is available
    - for example: "<funding>DOE:AC05-00OR22725</funding>"
  - does metadig-engine need login credentials for the awards lookup?
     - https://pamspublic.science.energy.gov/WebPAMSExternal/Interface/Awards/AwardSearchExternal.aspx?controlName=ContentTabs
     - didn't find a couple of award numbers that I used
- File formats - non-proprietary formats
  - does the proprietary format list need to be updated for ESS-DIVE
  - https://github.com/NCEAS/metadig-checks/blob/main/data/DataONEformats.csv
- File Present
  - is this the same as 'check.entity.present'?
    - https://github.com/NCEAS/metadig-checks/issues/126

**Comment by vchendrix** [comment]
Fuzzy Matches in R: https://cran.r-project.org/web/packages/stringdist/stringdist.pdf

**Comment by JEDamerow** [comment]
@gothub For the resource.URLs.resolvable check - can we add that the output includes up to 3 specific urls that were not resolvable?

For resource.projectTitle.controlled - when there is no exact match for the entered project name, add suggested method for users to find project name (Emily to send text for this)

**Comment by JEDamerow** [comment]
@gothub we decided to include a funder check to make sure that at least one specific funding source is listed "U.S. DOE > Office of Science > Biological and Environmental Research (BER)" Adding a note in that issue.

**Comment by JEDamerow** [comment]
We did a spot check today of some of the assessment reports, and found a few issues to fix, or discuss.  

[Notes on spot check issues encountered, with screenshots.](https://docs.google.com/document/d/1MRAN5I9y-tNieYcdvS06V2EtspUWzTPzQ-kBq6uZVvY/edit)
[List of datasets that we checked, with notes on any specific issues](https://docs.google.com/spreadsheets/d/1L_PWBpfn-ZLD1me4dLEBKH6CiHQYqmkSyw6DlDr9Hec/edit#gid=0)

### Summary of Issues:  

**Proprietary File check**
Showing up correctly, but may want to add a link to documentation describing what to do for non-proprietary formats (FOR ESS-DIVE TEAM TO DO)
EXCEL FILES - Common use case will be for excel files, and we have provided guidance that if you have/need to include an excel file, also include a csv version
PROPOSED SOLUTION: Have a second follow up check for excel files - If there is an excel file, check for matching csv.  OR have a message for all excel files “If you have not already, please also upload a csv version of your excel file”  

**Project Name check**
Change to an optional check to show up under warnings, and remove “Warning:” from the beginning of response


**Private Datasets** - “An Identifier”, and not running checks
Identifier issues -  “An identifier” is not present, “metadata identifier” not resolving
Fail message: “An identifier is not present.” What does this mean?  
One private dataset (​​ess-dive-f718f1571e4ac29-20220108T005614833) that did not run checks, and had 21 informational checks with “dialect for the check is not supported”


**Private dataset - Metadata identifier**
https://data.ess-dive.lbl.gov/view/ess-dive-ecc15a4f31c86ef-20211221T212550778
Failed check response: “The metadata identifier [ess-dive-ecc15a4f31c86ef-20211221T212550778](https://data.ess-dive.lbl.gov/view/ess-dive-ecc15a4f31c86ef-20211221T212550778)’ was found), but is not resolvable”
This is an ESS-DIVE issue right? Move to optional?  
Remove the “)” after found

**URL in metadata**
ISSUE: [http://localhost:3000/view/doi:10.15485/1631979](https://data.ess-dive.lbl.gov/view/doi:10.15485/1631979) … URL in metadata did not resolve correctly, because there was a comma (from sentence in abstract)
SOLUTION: Need to parse that out. Actually does resolve when you remove the comma from the end


**Funding Organization**
Failed check message: “The funding organization BER was not found in the metadata” 
Change to optional, listed in warnings

**Comment by gothub** [comment]
@JEDamerow @emilyarobles thanks for the review. 

Regarding your comments:

**Proprietary File check**
The proposed changes can be made, but we will have to create a new check that is unique to the ESS-DIVE suite instead of using the same check that is currently used by the FAIR suite.
The second check can be added (if excel file, check for matching csv), as an additional check.

*TODO*: Need to determine a name for this new check

**Project Name check**
This check will be changed to "OPTIONAL"

**Private Datasets** - “An Identifier”, and not running checks
> Identifier issues -  “An identifier” is not present, “metadata identifier” not resolving
> Fail message: “An identifier is not present.” What does this mean? 

This check simply looks for '/eml/@packageId' which is the identifier associated with the metadata document. Which dataset had this error?

> One private dataset (​​ess-dive-f718f1571e4ac29-20220108T005614833) that did not run checks, and had 21 informational checks with “dialect for the check is not supported”

This was due to a processing error that has been resolved, and shouldn't happen again.

**Private dataset - Metadata identifier**
> https://data.ess-dive.lbl.gov/view/ess-dive-ecc15a4f31c86ef-20211221T212550778
> Failed check response: “The metadata identifier [ess-dive-ecc15a4f31c86ef-20211221T212550778](https://data.ess->dive.lbl.gov/view/ess-dive-ecc15a4f31c86ef-20211221T212550778)’ was found), but is not resolvable”
> This is an ESS-DIVE issue right? Move to optional?  

The metadig engine has privilege to read private datasets, but the checks themselves run as unprivileged, so the call in the check to see if the URL HTTP 'Head' request is successful will not succeed for private datasets. This is really an NCEAS problem, and I will log an issue for it. It may be helpful to provide a more useful message if an HTTP 401 (Not authorized) message is returned. The check would have to be updated to detect this.

> Remove the “)” after found

will do, thx.

**URL in metadata**
> ISSUE: [http://localhost:3000/view/doi:10.15485/1631979](https://data.ess-dive.lbl.gov/view/doi:10.15485/1631979) … URL in metadata did not resolve correctly, because there was a comma (from sentence in abstract)
> SOLUTION: Need to parse that out. Actually does resolve when you remove the comma from the end

yes, that is indeed the problem, the check will be fixed.

**Funding Organization**
> Failed check message: “The funding organization BER was not found in the metadata” 
> Change to optional, listed in warnings.

OK, the check will be updated.

Please let me know if we need to discuss these further, or if any of your questions haven't been answered sufficiently.

**Comment by gothub** [comment]
@Val @JEDamerow issues reported with the new checks have been resolved and the ESS-DIVE 1.1.0 suite has been run for all current metadata. 

However, for at least one metadata document, there is an issue with the mediaType (formatId) associated with the EML entity. For example, for 'https://data.ess-dive.lbl.gov/view/ess-dive-fe751bc7ce7851b-20210930T015111122080':
```
    <otherEntity id="ess-dive-b4c4a1e743e0add-20190514T142445934">
      <entityName>PLOT_19_DPH_Response_Variables_20150814.csv</entityName>
      <entityType>application/vnd.ms-excel</entityType>
    </otherEntity>
```

The sysmeta for [this entry](https://data.ess-dive.lbl.gov/catalog/d1/mn/v2/meta/ess-dive-d8094078771d396-20190514T142445866) has the `formatId` as:
```
<formatId>application/vnd.ms-excel</formatId>
```

I downloaded the file, and it is definitely a CSV, so both the metadata and sysmeta should identify this as `text/csv`.

There may be other metadata/sysmeta that have this same problem. 

The check works as it is supposed to, as it uses the mediaType recorded in the metadata.

I'll continue to look for other pids that have this problem.
